corehq/ex-submodules/couchforms/geopoint.py

```diff
@@ -52,13 +52,14 @@ def _extract_elements(input_string, flexible):
 
 
 def _to_decimal(n):
-    """Coerces to Decimal and rounds small numbers to 0"""
+    """Coerces to Decimal and rounds small scientific notation inputs to 0"""
     try:
         ret = Decimal(n)
     except InvalidOperation as e:
         raise _GeoPointGenerationError(f"{n} is not a valid Decimal") from e
 
-    if not ret.is_nan() and abs(ret) < Decimal("0.001"):
+    uses_sci_notation = 'e' in n.lower()
+    if uses_sci_notation and abs(ret) < Decimal("0.001"):
```

Comment:
Why only coerce scientific notation inputs? Surely a normal input below 0.001 is also meaningless?

Reply:
That's what I did at first, but it loses precision.  Some of the existing tests use values of `0.0`, which would be turned into `0`.  It doesn't matter for elasticsearch, where the raw value is also stored, but I'm hesitant to change that behavior for jsonobject.  That is, if you put in `0.0`, that's what you should get out.